### ISSUE TITLE
loading image array Still having problems

### ISSUE THREAD

**Issue by danielcamara** [OlafenwaMoses/ImageAI#386]
Hi Thanks for the library, it is quite interesting! It hides a lot of problems and make it easy to use and train new models. Really nice work! 
However I am trying to load the numpy array instead the file, and I am having some problems! I already saw the problem #49, but it didn't solve for me. 

from PIL import Image
import numpy as np
image_arr = np.asarray(Image.open('/tmp/2.jpg'), dtype='float64')
detections = detector.detectObjectsFromImage(input_type="array", input_image=image_arr)

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/imageai/Detection/__init__.py", line 347, in detectObjectsFromImage
    _, _, detections = model.predict_on_batch(np.expand_dims(image, axis=0))
  File "/usr/local/lib/python3.6/dist-packages/keras/engine/training.py", line 1273, in predict_on_batch
    self._make_predict_function()
  File "/usr/local/lib/python3.6/dist-packages/keras/engine/training.py", line 554, in _make_predict_function
    **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/keras/backend/tensorflow_backend.py", line 2744, in function
    return Function(inputs, outputs, updates=updates, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/keras/backend/tensorflow_backend.py", line 2546, in __init__
    with tf.control_dependencies(self.outputs):
  File "/home/daniel/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 5426, in control_dependencies
    return get_default_graph().control_dependencies(control_inputs)
  File "/home/daniel/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4867, in control_dependencies
    c = self.as_graph_element(c)
  File "/home/daniel/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3796, in as_graph_element
    return self._as_graph_element_locked(obj, allow_tensor, allow_operation)
  File "/home/daniel/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3875, in _as_graph_element_locked
    raise ValueError("Tensor %s is not an element of this graph." % obj)
ValueError: Tensor Tensor("regression_1/concat:0", shape=(?, ?, 4), dtype=float32) is not an element of this graph.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py", line 3325, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-25-cc2c95978f97>", line 1, in <module>
    detections = detector.detectObjectsFromImage(input_type="array", input_image=image_arr)
  File "/usr/local/lib/python3.6/dist-packages/imageai/Detection/__init__.py", line 544, in detectObjectsFromImage
    "Ensure you specified correct input image, input type, output type and/or output image path ")
ValueError: Ensure you specified correct input image, input type, output type and/or output image path 

I am using: 
 -  Python 3.6 
 - ImageAI 2.1.5

Even though for me the ideal would be to use the open from scikit. It allows you to read files and
urls the same way... so it is more convenient, you don't need to get worried if it is a file or an url. 

from skimage import io 
import numpy as np
url='https://file.kbb.com/kbb/images/car-research/researchheroimage-retina.png'
image=io.imread(url)
image_arr = np.asarray(image, dtype='float64')
detections = detector.detectObjectsFromImage(input_type="array", input_image=image_arr)

basically gives me the same error!!
Do you have any idea?

**Comment by rola93** [comment]
Try modifying this line:
`image_arr = np.asarray(Image.open('/tmp/2.jpg'), dtype=np.uint8)`

**Comment by danielcamara** [comment]
FOUND the reason, you get that if you don't set the "output_image_path" to a real file!
In my case I don't need the output_image_path at all, even if I point it to /dev/null I got the error! 
I looked into your code, and by default the output is a file, if I don't provide a file name for the 
output everything breaks. 
This work though, changing the input type to array. 

from skimage import io as sk_io
from imageai.Detection import ObjectDetection
from imageai.Prediction import ImagePrediction

# models directory path
modelsPath = "."

# Detection model, loaded just once for performance purposes
detector = ObjectDetection()
detector.setModelTypeAsRetinaNet()

detector.setModelPath( os.path.join(modelsPath, "resnet50_coco_best_v2.0.1.h5"))
detector.loadModel()

detectedValues=detector.detectObjectsFromImage(input_type="array",
                                                                 input_image=image, output_type="array")
print(detectedValues)
-----
Maybe could be a good idea of improvement to the library could be to test if the input_type is "file", and no output_image_path is provided, return a more explicit error message. 

Again thanks for the library and rola93 for the comment :)

**Comment by rola93** [comment]
> Maybe a good idea of improvement to the library, could be to add a test if the input_type is "file", and no output_image_path is provided, return a more explicit error message.

Great idea! for sure there are things to improve here. Thanks for your time!
